### PR TITLE
BIOS: add new case in os firmware element

### PR DESCRIPTION
**Pull request by meinaLi**
1. Add new case in os firmware element
2. Fix an error in with_nvram_template but not with_template
Signed-off-by: Meina Li <meili@.redhat.com>

### CODE REVIEW

**Comment by meinaLi** [comment]
(p3-ve) [root@hp-compaq6005-01 tests]# avocado run --vt-type libvirt --vt-machine-type q35 virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev..no_os_loader
JOB ID     : b31fc57cf66bbbe2e1320ef140ab5dc837739594
JOB LOG    : /root/avocado/job-results/job-2019-09-18T04.18-b31fc57/job.log
 (1/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.os_nvram.valid_template.no_os_loader.without_secure: PASS (51.27 s)
 (2/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.os_nvram.valid_template.no_os_loader.with_secure: PASS (54.58 s)
 (3/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.os_nvram.no_template.no_os_loader.without_secure: PASS (54.67 s)
 (4/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.os_nvram.no_template.no_os_loader.with_secure: PASS (55.35 s)
 (5/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.no_os_nvram.no_os_loader.without_secure: PASS (54.61 s)
 (6/6) type_specific.io-github-autotest-libvirt.virsh.boot.by_ovmf.positive_test.options.boot.hd.file_disk.boot_dev.no_os_nvram.no_os_loader.with_secure: PASS (54.39 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 328.52 s

**Comment by meinaLi** [comment]
This PR mainly depend on a new avocado-vt PR https://github.com/avocado-framework/avocado-vt/pull/2270 to use. So both of them need to be merged together.

File: libvirt/tests/src/bios/virsh_boot.py [hunk 1]
```diff
@@ -694,6 +715,8 @@ def run(test, params, env):
         if ceph_cfg:
             os.remove(ceph_cfg)
         logging.debug("Start to cleanup")
+        if os.path.exists("/etc/qemu/"):
+            process.run("rm -rf /etc/qemu/", shell=True, ignore_status=True)
```
Comment (lento-sun) [inline libvirt/tests/src/bios/virsh_boot.py]: Suggest only remove the dirs/files if they are created by the script.
e.g. if some people or some other application created the same name dir in his/her env, and run this script, the dir will be lost after execution.

File: libvirt/tests/src/bios/virsh_boot.py [hunk 2]
```diff
@@ -308,6 +311,24 @@ def apply_boot_options(vmxml, params):
...
+        else:
+            logging.debug("Set os firmware:")
+            dict_os_attrs.update({"os_firmware": os_firmware})
+            # Need to prepare a zero-length file named "40-edk2-ovmf-sb.json" to hide it
+            # when test SB disabled guest (BZ 1564270)
+            file_path = "/etc/qemu/firmware/"
+            hidden_file = os.path.join(file_path, "40-edk2-ovmf-sb.json")
+            if not os.path.exists(file_path):
+                os.makedirs(file_path)
+                cmd = 'cat /dev/null >%s' % hidden_file
```
Comment (chunfuwen) [inline libvirt/tests/src/bios/virsh_boot.py]: touch command is more common for creating empty file

File: libvirt/tests/src/bios/virsh_boot.py (hunk 1 above)
Comment (meinaLi) [reply libvirt/tests/src/bios/virsh_boot.py]: Updated

File: libvirt/tests/src/bios/virsh_boot.py (hunk 2 above)
Comment (meinaLi) [reply libvirt/tests/src/bios/virsh_boot.py]: updated

**Pull request merged**